tests/cc/test_usdt_args.cc

```diff
@@ -54,9 +54,11 @@ static void verify_register(USDT::ArgumentParser &parser, int arg_size,
 
 TEST_CASE("test usdt argument parsing", "[usdt]") {
   SECTION("parse failure") {
-#ifdef __powerpc64__
+#ifdef __aarch64__
```

Comment:
Theoritically these tests doesn't rely on architecture. We just put in fake argument format strings and see if the parser outputs correct results. It would be nice to always test all architectures all the time, right?

Reply:
Okay. I will add the arch guard back to the test.

Reply:
> Theoritically these tests doesn't rely on architecture. We just put in fake argument format strings, call specific parser, and see if it outputs correct results. It would be nice to always test all architectures all the time, right?

The test _input_ needs to be different for different architecture. That is why we have different "ifdef"s.

Reply:
It feels to me that, say, the code `ArgumentParser_aarch64` itself, can also run on x86, is that right?

In that case, we can always invoke `ArgumentParser_aarch64` on arm64 test inputs, `USDT::ArgumentParser_powerpc64` on Power PC inputs, etc.?

Otherwise, since our testbot only runs on x86, we won't be able to catch regressions and breakage for other architectures at PR time?

Reply:
I still prefer that it is guarded by different architecture. The reason is usdt parsing code is arch specific. Even if we run some other arch inputs on x86, this still won't catch regression
on powerpc since powerpc parsing code is not exercised on x86.

There is a chance that illegal input in one arch is legal in a different one.
For example 4@4 is illegal on x86 but it is legal in aarch64. Mixing together
will prevent put 4@4 in negative test since it is legal in aarch64.

Reply:
After the second thought, I know what you mean. Yes, we could exercise non x86 parsing code on x86 host. I will make a revision to reflect that.

Reply:
Sorry, I cannot make the change because powerpc64 uses the regex library which old compiler likes gcc 4.8.5 does not have a good support (as I mentioned in my commit message as well).

....
 REQUIRE( !parser.parse(&arg) )
due to unexpected exception with message:
  regex_error
....

Therefore, let us stick to arch. specific usdt parser tests for now. Supposedly, each arch parser change should have run this test on their respective arch to catch all possible errors.